### PR TITLE
[GEARPUMP-312] Add Message trait and DefaultMessage impl

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [x] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by huafengw** [comment]
Can we and should we add an apply function in trait Message's companion object which returns a `DefaultMessage`?

**Comment by huafengw** [comment]
Document update?

**Comment by manuzhang** [comment]
@huafengw  updated

File: integrationtest/core/src/it/scala/org/apache/gearpump/integrationtest/checklist/DefaultMessageDeliverySpec.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ import org.apache.gearpump.integrationtest.{TestSpecBase, Util}
 /**
  * Checks message delivery consistency, like at-least-once, and exactly-once.
  */
-class MessageDeliverySpec extends TestSpecBase {
+class DefaultMessageDeliverySpec extends TestSpecBase {
```
Comment (huafengw) [inline integrationtest/core/src/it/scala/org/apache/gearpump/integrationtest/checklist/DefaultMessageDeliverySpec.scala]: Do we need to change this name?

Comment (manuzhang) [inline integrationtest/core/src/it/scala/org/apache/gearpump/integrationtest/checklist/DefaultMessageDeliverySpec.scala]: oh, my IDE is to blame for this.

**Comment by huafengw** [comment]
+1

**Pull request closed without merging**